core/com.b2international.snowowl.core.rest/src/com/b2international/snowowl/core/rest/codesystem/CodeSystemRestService.java

```diff
@@ -58,14 +70,88 @@
 	private CodeSystemService codeSystemService;
 	
 	@ApiOperation(
-			value="Retrieve all code systems",
-			notes="Returns a list containing generic information about registered code systems.")
+		value="Retrieve Code Systems", 
+		notes="Returns a collection resource containing all/filtered registered Code Systems."
+			+ "<p>Results are always sorted by repositoryUuid first, sort keys only apply per repository."
+			+ "<p>The following properties can be expanded:"
+			+ "<p>"
+			+ "&bull; availableUpgrades() &ndash; a list of possible code system URIs that can be used as an 'extensionOf' property"
+	)
 	@ApiResponses({
-		@ApiResponse(code = 200, message = "OK")
+		@ApiResponse(code = 200, message = "OK", response = CodeSystems.class),
+		@ApiResponse(code = 400, message = "Invalid search config", response = RestApiError.class),
+		@ApiResponse(code = 404, message = "Branch not found", response = RestApiError.class)
 	})
 	@GetMapping(produces = { AbstractRestService.JSON_MEDIA_TYPE })
-	public CollectionResource<CodeSystem> getCodeSystems() {
-		return CollectionResource.of(codeSystemService.getCodeSystems());
+	public @ResponseBody Promise<CodeSystems> searchByGet(final CodeSystemRestSearch params) {
```

Comment:
The responsebody annotation can be removed, because the class is annotated with `RestController`.
```suggestion
	public Promise<CodeSystems> searchByGet(final CodeSystemRestSearch params) {
```

Reply:
The same applies to the other methods.